### PR TITLE
Examining Rwalls now tells you which tool is needed to continue deconstructing it

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
:cl: Kor
add: You can now examine an r-wall to find out which tool is needed to continue deconstructing it.
/:cl:

### CODE REVIEW

**Comment by ma44** [comment]
Time to do this with mechs and regular walls and windows, cause i got no knowledge

**Comment by pudl** [comment]
Why does everything have to be perfectly placed in front of you? Isnt the mini-difficulty curve that SS13 has part of the appeal??????

**Comment by KorPhaeron** [comment]
Having a wiki page open and tabbing between them isn't part of the appeal for me, no.

**Comment by pudl** [comment]
Its not like the steps are nonsensical and random so you have to see the wiki every time to know what they are, you can infer whats next once youve done it a few times with the wiki

**Comment by PKPenguin321** [comment]
>you plebs need the wiki

get
good?

**Comment by PKPenguin321** [comment]
nah but this is cool, i recall when i first started playing this was one of my most requested features since i always forgot the order. super good for new players and old players can just not use it if they don't want

**Comment by TheCarlSaganExpress** [comment]
I think this is an excellent change. There's really not much of a skill curve for tearing down a wall since you can just look at the wiki or even write the process down on a piece of paper. Maybe we'll see a bunch of new players tearing down walls for fun, but sec could always just arrest them.

**Comment by Aranclanos** [comment]
i disagree with cheri

**Comment by AdamElTablawy** [comment]
like i mean sure there's a fun part of a game where like 'ooh undiscovered feature' where things aren't outlined on the wiki and you figure it out as you go along

but another thing is that the game is almost completely unplayable without SOME form of guideline and tbh being able to decon an rwall as a complete newbie without searching how to is not going to trash the game

**Comment by ike709** [comment]
I agree that the majority of wiki things can usually be remembered after doing it a few times, but rwalls is something that always gets me. I always mess up the order. 👍

**Comment by kevinz000** [comment]
Hugbox

**Comment by MMMiracles** [comment]
I feel like it'd be a lot better if it was more of a visual description instead of out-right telling you what was needed done.

So instead of
 >The outer grille has been removed. A screwdriver could remove the support lines.

You get
>The outer grille appears cut apart, revealing screws holding in the support lines.

It gets the same message to the player about what they should do without it be blatantly 'do this and that to get to next step'.

**Comment by TechnoAlchemisto** [comment]
basically what MMMIracles said, verbal queues are good since our graphics aren't enough to support figuring things out in terms of deconstructing. But we shouldn't just give it to 'em.

**Comment by AnturK** [comment]
I  agree that clearer progression messages would probably feel more natural than examine message.

**Comment by nullbear** [comment]
damn, you beat me to it.

Can you at least fix the bug to do with repairing them?

You can only repair it during the second step of deconstruction.

File: code/game/turfs/simulated/walls_reinforced.dm [hunk 1]
```diff
@@ -97,7 +125,7 @@
 					return 1
 				return 1
```
Comment (nullbear) [inline code/game/turfs/simulated/walls_reinforced.dm]: You can only repair the wall during the second step of the deconstruction process.

Error message is extremely niche and would be better placed within sheet/use() code.

**Review by nullbear — CHANGES_REQUESTED** [review]

**Comment by WJohn** [comment]
👍 
There's no need to be uselessly obscure to new players. This doesn't affect veterans or game balance, and there's no reason to fight this change. Calling this "hugbox" means you're kind of an idiot. It's  basic user feedback, which this game is still sorely lacking.

**Comment by Giacom** [comment]
I agree with the change but as MMMiracles mentioned instead of "do this next" it should say something like "the wires are exposed".

**Comment by Jalleo** [comment]
@Giacom that would only make sense if it was done consistently across the board to everything that can be deconstructed. and even then it will still increase the learning curve

**Comment by Giacom** [comment]
Pretty sure almost everything already does the "The wires are exposed" style descriptions instead of just saying "Wires are exposed, use a wirecutter".

I also don't think it would increase the learning curve if you use well thought-out descriptions. Unless you are retarded, I think a player can work out a description saying "There are bolts holding down the plating." means it can be wrenched.

**Comment by feemjmeem** [comment]
I agree about expanding the descriptions. It improves LOOOOOORRRRREEEE and any guidance is better than no guidance: descriptive hints aren't needlessly obtuse when there aren't any hints at all right now.

**Comment by nullbear** [comment]
I heavily agree with @Giacom 's feedback. When i did this (forgot to make a PR for it, you literally beat me by a hair damnit!) this was my examine proc:

```
/turf/closed/wall/plasteel/examine(mob/user)
	..()
	switch(decon_step)
		if(0)
			user << "<span class='notice'>The outer grille is fully intact.</span>"
		if(1)
			user << "<span class='notice'>The outer grille has been cut, and the support lines are screwed securely to the outer cover.</span>"
		if(2)
			user << "<span class='notice'>The support lines have been unscrewed, and the metal cover is welded firmly in place.</span>"
		if(3)
			user << "<span class='notice'>The metal cover has been sliced through, and is connected loosely to the girder.</span>"
		if(4)
			user << "<span class='notice'>The outer cover has been pried away, and the bolts anchoring the support rods are wrenched in place.</span>"
		if(5)
			user << "<span class='notice'>The bolts anchoring the support rods have been loosened, but are still welded firmly to the girder.</span>"
		if(6)
			user << "<span class='notice'>The support rods have been sliced through, and the outer sheath is connected loosely to the girder.</span>"
```

On another note, i also fixed the 'unable to repair' bug, and made repairing it use the previous step, instead of some metal (?!?!?? which was never used to construct the wall in the first place??? ITS an r_Wall??)

Here is my code: Keep in mind, there are unrelated changes related to my materials refactor:

```
	switch(decon_step)
		if(0) // Pristine Cosmetic Condition
			if(istype(I, /obj/item/weapon/wirecutters))
				playsound(src, 'sound/items/Wirecutter.ogg', 100, 1)
				decon_step = 1
				update_icon()
				user << "<span class='notice'>You cut the outer grille.</span>"
				max_integrity -= 50 // 450
				integrity = min(integrity, max_integrity)
				return 1

		if(1) // Grille has been cut.
			if(istype(I, /obj/item/weapon/screwdriver))
				user << "<span class='notice'>You begin removing the support lines...</span>"
				playsound(src, 'sound/items/Screwdriver.ogg', 100, 1)
				var/turf/T = user.loc
				if(do_after(user, 40/I.toolspeed, target = src))
					if(!istype(src, /turf/closed/wall/plasteel) || !user || !I || !T || user.loc != T || user.get_active_held_item() != I || decon_step != 1)
						return 1
					decon_step = 2
					update_icon()
					user << "<span class='notice'>You remove the support lines.</span>"
					max_integrity -= 50 // 400
					integrity = min(integrity, max_integrity)
				return 1

			if(istype(I, /obj/item/weapon/wirecutters))
				playsound(src, 'sound/items/Wirecutter.ogg', 100, 1)
				decon_step = 0
				update_icon()
				user << "<span class='notice'>You repair the outer grille.</span>"
				max_integrity += 50 // 500
				integrity = min(integrity+25, max_integrity)
				return 1

		if(2) // Support lines are removed.
			if(istype(I, /obj/item/weapon/weldingtool))
				var/obj/item/weapon/weldingtool/WT = I
				if(WT.remove_fuel(0,user))
					user << "<span class='notice'>You begin slicing through the metal cover...</span>"
					playsound(src, 'sound/items/Welder.ogg', 100, 1)
					var/turf/T = user.loc
					if(do_after(user, WALL_WELD_TIME(integrity)/WT.toolspeed, target = src))
						if(!istype(src, /turf/closed/wall/plasteel) || !user || !WT || !T || !WT.isOn() || user.loc != T || user.get_active_held_item() != WT || decon_step != 2)
							return 1
						decon_step = 3
						update_icon()
						user << "<span class='notice'>You press firmly on the cover, dislodging it.</span>"
						max_integrity -= 50 // 350
						integrity = min(integrity, max_integrity)
				return 1

			if(istype(I, /obj/item/weapon/screwdriver))
				user << "<span class='notice'>You begin securing the support lines...</span>"
				playsound(src, 'sound/items/Screwdriver.ogg', 100, 1)
				var/turf/T = user.loc
				if(do_after(user, 40/I.toolspeed, target = src))
					if(!istype(src, /turf/closed/wall/plasteel) || !user || !I || !T || user.loc != T || user.get_active_held_item() != I || decon_step != 2)
						return 1
					decon_step = 1
					update_icon()
					user << "<span class='notice'>The support lines have been re-secured.</span>"
					max_integrity += 50 // 450
					integrity = min(integrity+25, max_integrity)
				return 1

		if(3) // Outer cover sliced
			if(istype(I, /obj/item/weapon/crowbar))
				user << "<span class='notice'>You struggle to pry off the cover...</span>"
				playsound(src, 'sound/items/Crowbar.ogg', 100, 1)
				var/turf/T = user.loc
				if(do_after(user, 100/I.toolspeed, target = src))
					if(!istype(src, /turf/closed/wall/plasteel) || !user || !I || !T || user.loc != T || user.get_active_held_item() != I || decon_step != 3)
						return 1
					decon_step = 4
					update_icon()
					user << "<span class='notice'>You pry off the cover.</span>"
					max_integrity -= 50 // 300
					integrity = min(integrity, max_integrity)
				return 1

			if(istype(I, /obj/item/weapon/weldingtool))
				var/obj/item/weapon/weldingtool/WT = I
				if(WT.remove_fuel(0,user))
					user << "<span class='notice'>You begin welding the metal cover...</span>"
					playsound(src, 'sound/items/Welder.ogg', 100, 1)
					var/turf/T = user.loc
					if(do_after(user, WALL_WELD_TIME(integrity)/WT.toolspeed, target = src))
						if(!istype(src, /turf/closed/wall/plasteel) || !user || !WT || !T || !WT.isOn() || user.loc != T || user.get_active_held_item() != WT || decon_step != 3)
							return 1
						decon_step = 2
						update_icon()
						user << "<span class='notice'>The metal cover has been welded securely.</span>"
						max_integrity += 50 // 400
						integrity = min(integrity+25, max_integrity)
				return 1

		if(4) // Outer Cover removed.
			if (istype(I, /obj/item/weapon/wrench))
				user << "<span class='notice'>You start loosening the anchoring bolts which secure the support rods to their frame...</span>"
				playsound(src, 'sound/items/Ratchet.ogg', 100, 1)
				var/turf/T = user.loc
				if(do_after(user, 40/I.toolspeed, target = src))
					if(!istype(src, /turf/closed/wall/plasteel) || !user || !I || !T || user.loc != T || user.get_active_held_item() != I || decon_step != 4)
						return 1
					decon_step = 5
					update_icon()
					user << "<span class='notice'>You remove the bolts anchoring the support rods.</span>"
					max_integrity -= 50 // 250
					integrity = min(integrity, max_integrity)
				return 1

			if(istype(I, /obj/item/weapon/crowbar))
				user << "<span class='notice'>You start to pry the cover back into place...</span>"
				playsound(src, 'sound/items/Crowbar.ogg', 100, 1)
				var/turf/T = user.loc
				if(do_after(user, 20/I.toolspeed, target = src))
					if(!istype(src, /turf/closed/wall/plasteel) || !user || !I || !T || user.loc != T || user.get_active_held_item() != I || decon_step != 4)
						return 1
					decon_step = 3
					update_icon()
					user << "<span class='notice'>The metal cover has been pried back into place.</span>"
					max_integrity += 50 // 350
					integrity = min(integrity+50, max_integrity)
				return 1

		if(5) // Support rods loosened.
			if(istype(I, /obj/item/weapon/weldingtool))
				var/obj/item/weapon/weldingtool/WT = I
				if(WT.remove_fuel(0,user))
					user << "<span class='notice'>You begin slicing through the support rods...</span>"
					playsound(src, 'sound/items/Welder.ogg', 100, 1)
					var/turf/T = user.loc
					if(do_after(user, WALL_WELD_TIME(integrity)/WT.toolspeed, target = src))
						if(!istype(src, /turf/closed/wall/plasteel) || !user || !WT || !T || !WT.isOn() || user.loc != T || user.get_active_held_item() != WT || decon_step != 5)
							return 1
						decon_step = 6
						update_icon()
						user << "<span class='notice'>You slice through the support rods.</span>"
						max_integrity -= 50 // 200 // It's now just as strong as a normal wall.
						integrity = min(integrity, max_integrity)
				return 1

			if (istype(I, /obj/item/weapon/wrench))
				user << "<span class='notice'>You start tightening the anchoring bolts which secure the support rods to their frame...</span>"
				playsound(src, 'sound/items/Ratchet.ogg', 100, 1)
				var/turf/T = user.loc
				if(do_after(user, 40/I.toolspeed, target = src))
					if(!istype(src, /turf/closed/wall/plasteel) || !user || !I || !T || user.loc != T || user.get_active_held_item() != I || decon_step != 5)
						return 1
					decon_step = 4
					update_icon()
					user << "<span class='notice'>You've tightened the bolts anchoring the support rods.</span>"
					max_integrity += 50 // 250
					integrity = min(integrity+25, max_integrity)
				return 1

		if(6) // Support rods sliced and removed.
			if(istype(I, /obj/item/weapon/crowbar))
				user << "<span class='notice'>You struggle to pry off the outer sheath...</span>"
				playsound(src, 'sound/items/Crowbar.ogg', 100, 1)
				var/turf/T = user.loc
				if(do_after(user, 100/I.toolspeed, target = src))
					if(!istype(src, /turf/closed/wall/plasteel) || !user || !I || !T || user.loc != T || user.get_active_held_item() != I || decon_step != 6)
						return 1
					user << "<span class='notice'>You pry off the outer sheath.</span>"
					decon_act(WRECKAGE_CHANCE_DAMAGE(integrity, max_integrity), I.damtype)
				return 1

			if(istype(I, /obj/item/weapon/weldingtool))
				var/obj/item/weapon/weldingtool/WT = I
				if(WT.remove_fuel(0,user))
					user << "<span class='notice'>You begin welding the support rods into place...</span>"
					playsound(src, 'sound/items/Welder.ogg', 100, 1)
					var/turf/T = user.loc
					if(do_after(user, WALL_WELD_TIME(integrity)/WT.toolspeed, target = src))
						if(!istype(src, /turf/closed/wall/plasteel) || !user || !WT || !T || !WT.isOn() || user.loc != T || user.get_active_held_item() != WT || decon_step != 6)
							return 1
						decon_step = 5
						update_icon()
						user << "<span class='notice'>The support rods have been welded securely.</span>"
						max_integrity += 50 // 200 // It's now just as strong as a normal wall.
						integrity = min(integrity+25, max_integrity)
				return 1
```

**Comment by KorPhaeron** [comment]
I'll let nullbear do it then I guess

**Comment by KorPhaeron** [comment]
damn it has conflicts nevermind too much effort

**Comment by nullbear** [comment]
xD am i taking too long?

**Comment by KorPhaeron** [comment]
yeah

**Comment by nullbear** [comment]
sry i was napping. I'll do some work.

**Pull request closed without merging**